solutions/usaco-941.mdx

```diff
@@ -2,9 +2,74 @@
 id: usaco-941
 source: USACO Bronze 2019 Open
 title: Cow Evolution
-author: Jesse Choe
+author: Jesse Choe, Ryan Chou
 ---
 
+## Solution 
 [Official Analysis](http://www.usaco.org/current/data/sol_evolution_bronze_open19.html)
 
 [Video Analysis](https://youtu.be/4YTijr0BbrA)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+<PySection>
+  
+```py
+import sys
+sys.stdin = open('evolution.in', 'r')
+sys.stdout = open('evolution.out', 'w')
+
+n = int(input())
+cows = []
+allchars = set()
+
+for i in range(n):
+	line = input().split()
+	chars = list(line[1:])
+	cows.append(chars)
+	allchars.update(chars)
+
+allchars = list(allchars)
+
+def solve():
+	# iterating through every pair of characteristics.
+	for i in range(len(allchars)):
+		for j in range(i + 1, len(allchars)):
+			if scan(i, j):
+				print('no')
+				return 0
+	print('yes')
+	return 0
+
+def scan(a, b):
+	""""
+	If we find a cow which has the characteristic a, 
```

Comment:
maybe smth like "check if the tree is evolutionarily proper relative to characteristics a and b" or something then add the original doc comment